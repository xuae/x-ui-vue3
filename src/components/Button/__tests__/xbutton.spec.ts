import { shallowMount } from '@vue/test-utils';
import XButton from '@/components/Button/XButton.vue';

describe('XButton.vue', () => {
  it('create', () => {
    const wrapper = shallowMount(XButton);
    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.classes('x-button')).toBe(true);
  });

  it('nativeType', async () => {
    const data = ['button', 'submit', 'reset'];
    for (let i = 0; i < data.length; i++) {
      const wrapper = shallowMount(XButton as any, {
        props: {
          nativeType: data[i],
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.attributes('type')).toBe(data[i]);
    }
  });

  it('stylus', async () => {
    const data = ['text', 'icon', 'outlined', 'contained', 'toggle'];
    for (let i = 0; i < data.length; i++) {
      const wrapper = shallowMount(XButton as any, {
        props: {
          stylus: data[i],
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes(`x-button__${data[i]}`)).toBe(true);
    }
  });

  it('type', () => {
    const data = ['flat', 'fab'];
    for (let i = 0; i < data.length; i++) {
      const wrapper = shallowMount(XButton as any, {
        props: {
          type: data[i],
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes(`x-button__${data[i]}`)).toBe(true);
    }
  });

  it('ripple', () => {
    const data = [true, false];
    for (let i = 0; i < data.length; i++) {
      const wrapper = shallowMount(XButton as any, {
        props: {
          ripple: data[i],
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('x-button__ripple')).toBe(data[i]);
    }
  });

  it('round', () => {
    const data = [true, false];
    for (let i = 0; i < data.length; i++) {
      const wrapper = shallowMount(XButton as any, {
        props: {
          round: data[i],
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('x-button__round')).toBe(data[i]);
    }
  });

  it('small', () => {
    const data = [true, false];
    for (let i = 0; i < data.length; i++) {
      const wrapper = shallowMount(XButton as any, {
        props: {
          small: data[i],
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('x-button__small')).toBe(data[i]);
    }
  });

  it('block', async () => {
    const data = [true, false];
    for (let i = 0; i < data.length; i++) {
      const wrapper = shallowMount(XButton as any, {
        props: {
          block: data[i],
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('x-button__block')).toBe(data[i]);
    }
  });

  it('disabled', () => {
    const data = [true, false];
    for (let i = 0; i < data.length; i++) {
      const wrapper = shallowMount(XButton as any, {
        props: {
          disabled: data[i],
        },
      });
      expect(wrapper.element).toMatchSnapshot();
      expect(wrapper.classes('x-button__disabled')).toBe(data[i]);
    }
  });
});
