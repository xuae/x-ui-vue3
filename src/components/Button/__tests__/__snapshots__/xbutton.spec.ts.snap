// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XButton.vue block 1`] = `
<button
  class="x-button   x-button__ripple x-button__block"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue block 2`] = `
<button
  class="x-button   x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue create 1`] = `
<button
  class="x-button   x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue disabled 1`] = `
<button
  class="x-button   x-button__ripple x-button__disabled"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue disabled 2`] = `
<button
  class="x-button   x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue nativeType 1`] = `
<button
  class="x-button   x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue nativeType 2`] = `
<button
  class="x-button   x-button__ripple"
  type="submit"
>
  Button
</button>
`;

exports[`XButton.vue nativeType 3`] = `
<button
  class="x-button   x-button__ripple"
  type="reset"
>
  Button
</button>
`;

exports[`XButton.vue ripple 1`] = `
<button
  class="x-button   x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue ripple 2`] = `
<button
  class="x-button"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue round 1`] = `
<button
  class="x-button   x-button__ripple x-button__round"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue round 2`] = `
<button
  class="x-button   x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue small 1`] = `
<button
  class="x-button   x-button__ripple x-button__small"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue small 2`] = `
<button
  class="x-button   x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue stylus 1`] = `
<button
  class="x-button x-button__text  x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue stylus 2`] = `
<button
  class="x-button x-button__icon  x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue stylus 3`] = `
<button
  class="x-button x-button__outlined  x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue stylus 4`] = `
<button
  class="x-button x-button__contained  x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue stylus 5`] = `
<button
  class="x-button x-button__toggle  x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue type 1`] = `
<button
  class="x-button  x-button__flat x-button__ripple"
  type="button"
>
  Button
</button>
`;

exports[`XButton.vue type 2`] = `
<button
  class="x-button  x-button__fab x-button__ripple"
  type="button"
>
  Button
</button>
`;
